_integration-tests/utils/agent/agent.go

```diff
@@ -0,0 +1,188 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+package agent
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/beevik/guid"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+)
+
+type MockAgent struct {
+	virtualEnv     string
+	process        *exec.Cmd
+	processCancel  context.CancelFunc
+	currentSession atomic.Pointer[mockSession]
+	port           int
+}
+
+type mockSession struct {
+	agent *MockAgent
+	token *guid.Guid
+}
+
+func New(t *testing.T) (agent MockAgent, err error) {
+	if agent.virtualEnv, err = os.MkdirTemp("", "orchestrion-integ-venv-*"); err != nil {
+		return
+	}
+	t.Logf("Creating Python venv at %q...\n", agent.virtualEnv)
+	if err = exec.Command("python3", "-m", "venv", agent.virtualEnv).Run(); err != nil {
+		return
+	}
+	venvBin := filepath.Join(agent.virtualEnv, "bin")
+	if runtime.GOOS == "windows" {
+		venvBin = filepath.Join(agent.virtualEnv, "Scripts")
+	}
+
+	t.Logf("Installing ddapm-test-agent in venv...\n")
+	if err = exec.Command(filepath.Join(venvBin, "pip"), "install", "ddapm-test-agent").Run(); err != nil {
+		return
+	}
```

Comment:
Requests Changes: I believe it makes more sense here to create a `.venv` folder in the current repository. We can then add it to the .gitignore and make it part of the CI cache.

Nit: We could also make this code part of simple `integration-tests.(sh|ps1)` ( :stuck_out_tongue_winking_eye: ) at the root that would run the test-agent and `go test -build-tags=integration`

Reply:
>Suggestion: We could also make this code part of simple `integration-tests.(sh|ps1)` ( 😜 ) at the root that would run the test-agent and go test `-build-tags=integration`

I specifically want to avoid having folks use anything other than `go` or `orchestrion go` commands, so I specifically want to NOT have such a script anymore.

Reply:
I've made it possible to use a "from-env" binary, and the CI tests cache that now...